tools/cldr-unittest/src/org/unicode/cldr/unittest/TestShim.java

```diff
@@ -2,19 +2,31 @@
 
 import java.io.PrintWriter;
 
+import org.unicode.cldr.util.TestShimUtils;
+
 import org.junit.jupiter.api.Test;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
 
 /**
  * a JUnit test that calls TestAll.
  */
 class TestShim {
     @Test
     public void TestAll() {
-        String args[] = { "-n", "-q" }; // TODO: parameterize
+        String args[] = TestShimUtils.getArgs(TestShim.class, "-n -q");
         // regular main() will System.exit() which is not too friendly.
         // call this instead.
         int errCount = TestAll.runTests(args);
         assertEquals(errCount, 0, "Test had errors");
     }
+
+    @Test
+    public void TestTestShimUtilTest() {
+        // Note: checks the system property corresponding to java.lang,
+        // we expect this to be unset!
+        String args[] = TestShimUtils.getArgs(java.lang.String.class, "-a -b -c");
```

Comment:
I don't understand the comment at all: "... system property corresponding to java.lang ... unset!"

Could it be clarified?

Reply:
Yes, i'll clarify. To test the mechanism, it assumes `java.lang.testArgs` is unset.